src/Input/Keybindings.rei

```diff
@@ -2,10 +2,7 @@ type t;
 
 let empty: t;
 
-type effect =
-  | Command(string)
-  | Text(string)
-  | Unhandled(EditorInput.KeyPress.t);
+type effect = EditorInput.effect(string);
```

Comment:
You don't have to expose the alias in the interface if you don't want to. It can remain an implementation detail as long as you provide the full definition in the implementation, in addition to the alias.

So If in the implementation you have:

```
type effect = EditorInput.effect(string) =
  | Command(string)
  | Text(string)
  | Unhandled(EditorInput.KeyPress.t);
```

Then you can leave the interface as:

```suggestion
type effect =
  | Command(string)
  | Text(string)
  | Unhandled(EditorInput.KeyPress.t);
``

and it won't be a breaking change for consumers.

Reply:
Hmm, I tried this, and I'm getting an error message:
```
# Unformatted Error Output:
# File "src/Input/Keybindings.re", line 11, characters 0-111:
# Error: This variant or record definition does not match that of type
#          string EditorInput.effect
#        They have different arities.


 ERROR  src/Input/Keybindings.re:11:0-14:35 

  8 ┆ 
  9 ┆ open EditorInput;
 10 ┆ 
 11 ┆ type effect = EditorInput.effect(string) =
 12 ┆ | Command(string)
 13 ┆ | Text(string)
 14 ┆ | Unhandled(EditorInput.KeyPress.t);
```

Reply:
In this commit: https://github.com/onivim/oni2/pull/1497/commits/6896763d1d93cdf0cea71ba11af90e1897052d9d

Perhaps I'm making a silly mistake? It would be nice to have this.

Reply:
Hmm, I guess this is a limitation of it being polymorphic. That's unfortunate.

Could you put it back in the functor and just instantiate that earlier? Or does that create other problems?

Otherwise it really isn't that important.

Reply:
> Could you put it back in the functor and just instantiate that earlier? Or does that create other problems?

I believe this would create other problems, because the functorized-module is completely internal. I'm not sure how the `rei` would pick up that value. I think it would work if the type was abstract (like `type effect;`), but not in the case where we want to specify variant.

> Or you could make this polymorphic as well I guess, although that would still be pretty weird from a consumer POV. 

Right, I was trying that out, but it seems pretty awkward. I think the `EditorInput.effect(string)` is the 'best' balance at the moment. The KeyBindings interface also uses `EditorInput.KeyPress.t`.

Reply:
> I'm not sure how the rei would pick up that value.

Why would it need to? That's what we're trying to avoid, isn't it? I'm thinking it'd be just like the commit you reference above, except you'd instantiate the functor before defining `effect`, so you can alias it to the now non-polymorphic `Input.effect`. The `rei` shouldn't need to change.

> The KeyBindings interface also uses EditorInput.KeyPress.t.

Right. You could probably alias and hide that type too of course, but again, no need to spend a lot of time on this. It's more just nice-to-have and of general interest I think.

Reply:
> I'm thinking it'd be just like the commit you reference above, except you'd instantiate the functor before defining effect, so you can alias it to the now non-polymorphic Input.effect. 

Ah, right - just tried it out and it seems to work 👍